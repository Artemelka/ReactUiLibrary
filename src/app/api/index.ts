import { getLabels, getLocales } from './localization';

export const API = {
    localization: {
        getLabels,
        getLocales
    }
};
