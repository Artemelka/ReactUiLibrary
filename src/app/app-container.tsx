import React, { Component } from 'react';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { routerMiddleware } from 'react-router-redux';
import { reducers } from './reducers';
import { TranslateProvider, translateMiddleware } from '../services/translate';
import { request } from '../services';
import { App } from './index';

const history = createBrowserHistory();
const store = createStore(reducers, composeWithDevTools(
    applyMiddleware(
        routerMiddleware(history),
        translateMiddleware,
        thunk
)));

export class AppContainer extends Component {
    fetchDictionary = () => request('/api/translate');

    render() {
        return (
            <Provider store={store}>
                <TranslateProvider fetchDictionary={this.fetchDictionary}>
                    <Router history={history}>
                        <App />
                    </Router>
                </TranslateProvider>
            </Provider>
        );
    }
}
