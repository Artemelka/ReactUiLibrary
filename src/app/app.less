.App {
    bottom: 0;
    //font-family: 'Fira Sans', sans-serif;
    left: 0;
    overflow: auto;
    position: fixed;
    right: 0;
    top: 0;
}
