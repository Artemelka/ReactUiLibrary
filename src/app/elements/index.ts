export { Anchor } from './anchor/Anchor';
export { Button } from './buttons/Button';
export { ButtonsGroup } from './button-group/ButtonGroup';
export { CardList } from './card-list/CardList';
export { Checkbox, Input, RadioButtons, Select, Textarea } from './inputs';
export { DropDownPanel } from './drop-down-panel/DropDownPanel';
export { DropDownDetails } from './drop-down-panel/DropDownDetails';
export { DropDownSummary } from './drop-down-panel/DropDownSummary';
export { IconModule } from './icon';
export { Modal, ModalSize } from './modal/Modal';
export { Progress } from './progress';
export { Text } from './text';
