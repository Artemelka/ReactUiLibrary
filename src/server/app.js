import express from 'express';
import bodyParser from 'body-parser';
import fileUpload from 'express-fileupload';
import reload from 'express-reload';
import mongoose from 'mongoose';
import { DB_URL, DB_URL_CUT, ServerParams, KEEP_ALIVE_TIMEOUT } from './constants';

const app = express();
const { PORT, HOST } = ServerParams;
const path = __dirname + '/routes/';
const listenCallback = error => error
    ? console.log('something bad happened', error)
    : console.log(`Listening on port ${PORT}!`);

app.use(bodyParser.json());
app.use(bodyParser.text());
app.use(fileUpload());
app.use(express.static('dist'));
app.use('/api', reload(path));

const appServer = app.listen(PORT, HOST, listenCallback);

mongoose.connect(DB_URL, {
    useNewUrlParser: true,
    useUnifiedTopology: true
})
    .then(() => console.log('Connected to DB!'))
    .catch(error => console.error(error));

appServer.keepAliveTimeout = KEEP_ALIVE_TIMEOUT;
