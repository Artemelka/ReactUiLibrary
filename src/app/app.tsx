import React from 'react';
import { Route } from 'react-router';
import { TestHomePage, LibraryPage } from './pages';

import './app.less';

export class App extends React.Component<{}, {}> {
    getDuplicate = () => {
        console.log('result');
    };

    handleClick = () => {
        this.getDuplicate();

        console.log('result');
    };

    render() {
        return (
            <div className="App">
                {/*<Route path="/" exact component={TestHomePage} />*/}
                <Route path="/" exact component={LibraryPage} />
                {/*<Route path="/library" component={LibraryPage} />*/}
                <button style={{zIndex: 100, position: 'relative'}} onClick={this.handleClick}>duplicate</button>
            </div>
        );
    }
}

