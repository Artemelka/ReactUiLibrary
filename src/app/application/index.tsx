import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { AppLoader } from 'components';
import { StoreInjectorProvider } from '@artemelka/redux-store-injector';
import { appStore, history } from './redux';
import { AppContainer } from './App';
import { mockAllRequest } from '../api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <StoreInjectorProvider store={appStore}>
                <ConnectedRouter history={history}>
                    <AppContainer />
                    <AppLoader />
                </ConnectedRouter>
            </StoreInjectorProvider>
        </Provider>
    );
};
