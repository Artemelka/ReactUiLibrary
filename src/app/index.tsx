import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { appStore } from './store';
import { history } from './app-history';
import { AppContainer } from './app';
import { AppLoader } from './components';
import { mockAllRequest } from './api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <ConnectedRouter history={history}>
                <AppContainer />
                <AppLoader />
            </ConnectedRouter>
        </Provider>
    );
};
