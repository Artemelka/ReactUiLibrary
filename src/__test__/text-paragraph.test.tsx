import React from 'react';
import { shallow } from 'enzyme';
import { TextParagraph } from '../app/elements';

const TEST_TEXT = 'test text';

describe('Test TextParagraph element', () => {
    test('Rendering snapshot', () => {
        const output = shallow(<TextParagraph>{TEST_TEXT}</TextParagraph>);

        expect(output).toMatchSnapshot();
    });
    test('Expect className modifier --upper', () => {
        const output = shallow(<TextParagraph upper >{TEST_TEXT}</TextParagraph>);

        expect(output.find('.Text-paragraph--upper').exists()).toBe(true);
    });
    test('Expect className modifier --bold', () => {
        const output = shallow(<TextParagraph bold >{TEST_TEXT}</TextParagraph>);

        expect(output.find('.Text-paragraph--bold').exists()).toBe(true);
    });
    test('Expect className modifier --light', () => {
        const output = shallow(<TextParagraph light >{TEST_TEXT}</TextParagraph>);

        expect(output.find('.Text-paragraph--light').exists()).toBe(true);
    });
    test('Expect console.warn with bold and light props', () => {
        const testMock = jest.fn();
        console.warn = testMock;
        const output = shallow(<TextParagraph light bold >{TEST_TEXT}</TextParagraph>);

        expect(testMock).toHaveBeenCalled();

    });
    test('Expect light modifier with bold and light props', () => {
        const output = shallow(<TextParagraph light bold >{TEST_TEXT}</TextParagraph>);

        expect(output.find('.Text-paragraph--light').exists()).toBe(true);
    });
});