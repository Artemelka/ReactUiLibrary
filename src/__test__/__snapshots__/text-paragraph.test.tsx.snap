// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextParagraph element Rendering snapshot 1`] = `
<p
  className="Text-paragraph"
  style={
    Object {
      "fontSize": "16px",
    }
  }
>
  test text
</p>
`;
